Sources/MessagingPush/MessagingPush.swift

```diff
@@ -129,3 +155,14 @@ public class MessagingPush: ModuleTopLevelObject<MessagingPushInstance>, Messagi
         implementation?.trackMetric(deliveryID: deliveryID, event: event, deviceToken: deviceToken)
     }
 }
+
+// Convenient way for other modules to access CustomerIOInstance as well as being able to mock CustomerIOInstance.
+public extension DIGraphShared {
+    var messagingPushInstance: MessagingPushInstance {
+        if let override: MessagingPushInstance = getOverriddenInstance() {
+            return override
+        }
+
+        return MessagingPush.shared
+    }
+}
```

Comment:
This is added to write automated tests for `opened` tracking. This is required because currently, `opened` metrics are reported in the `MessagingPush` class. 

I am not the biggest fan of this code, but I didn't want to expand scope of this PR by changing implementation logic. This code is the same implementation being used in `main` today.

Reply:
hmm thinking out load, `MessagingPush` shouldn't ideally need `CustomerIOInstance`... We should be utilizing the event bus and the implementation class of  CustomerIOInstance should be handling those.

Reply:
I think you're referring to the comment in the code snippet that includes `CustomerIOInstance`. If so, that is an outdated comment. 

```suggestion
// Convenient way for other modules to access instance as well as being able to mock instance in tests.
public extension DIGraphShared {
    var messagingPushInstance: MessagingPushInstance {
        if let override: MessagingPushInstance = getOverriddenInstance() {
            return override
        }

        return MessagingPush.shared
    }
}
```

I'll update the comment to remove the confusion.